### DISCUSSION TITLE
was wird zukünftig aus den Kapitelnamen ?

### DISCUSSION THREAD

**Discussion by alurio** [audiamus/AaxAudioConverter#106] [General]
Hallo Audioamus! 
Eine Frage hätte ich aktuell was die Zukunft der Software betrifft.
Du arbeitest ja an einem großen Update, dass ohne die Windows 10 App auskommen soll. Angesichts der Tatsache, dass  Audible 
 offenbar seltsamerweise die Win 10 App vom Markt nimmt eine sinnvolle Entscheidung. 

Was wird dann aber aus den von mir sehr geschätzten feature  Kapitelnamen benutzen ?  Die werden ja bisher aus einer Jason Datei, die mit der Win 10 App heruntergeladen wird bezogen. 
Werden auch künftige Versionen die Kapitelnamen  auslesen können ? Weiß man wo diese Informationen auf dem Server liegen, bzw. wie sie  bezogen werden können ?  Irgendwo muss die Infos ja sein, schließlich benutzt auch der Cloud Player kapitelnamen. 

Viele Grüße 

Alurio

**Comment by audiamus** [comment]
Es bleibt beim gewohnten, alle Metadaten der .json-Dateien stehen auch zukünftig zur Verfügung. 

Vorläufig aber wird ein Zwischenschritt erforderlich sein. Während der Entwicklung der Download-Komponente hat sich herausgestellt, dass für deren Integration in AAX Audio Converter sowohl der Datenfluss als auch die interne Klassenstruktur verändert werden müssen. Daher habe ich mich entschlossen, den Downloader zunächst als eigenständiges Programm zu gestalten. Das ist nun gestern - wirklich reiner Zufall - soweit fertig geworden, dass ich es in einer ersten Version veröffentlichen konnte:
https://github.com/audiamus/BookLibConnect

Dank der Arbeiten in der Open-Source-Gemeinde ist ja nicht nur das neue Verschlüsselungsverfahren verstanden worden - unter der Dateiendung .aaxc bekannt, sondern auch der Umgang mit der Audible-internen Web-Schnittstelle. Und über diese Schnittstelle kommen auch genau die Metadaten, die die Audible Win10-App als .json abgelegt hat.

In Hinblick auf die zukünftige Integration in AAX Audio Converter wird bereits jetzt eine kleine lokale SQLite-Datenbank angelegt, in der die Metadaten gespeichert werden. 

Der neue Downloader Book Lib Connect enthält eine Export-Option, die die volle Kompatibilität zum jetzigen AAX Audio Converter herstellt. Der Export erzeugt Pseudo-aax-Dateien und partiell gefüllte .json-Dateien, so wie AAX Audio Converter sie erwartet. Die json-Daten kommen dann aus der lokalen Datenbank.    

_Ich werde auch in den Foren noch dazu schreiben._
